lib/mongodb.js

```diff
@@ -1040,7 +1124,15 @@ MongoDB.prototype.autoupdate = function(models, cb) {
             if (options.background === undefined) {
               options.background = true;
             }
+          } else if (properties[p].type && properties[p].type.name === 'GeoPoint') {
+            var indexType =
+              typeof properties[p].index === 'string' ?
+                properties[p].index : '2dsphere';
           // If properties[p].index isn't an object we hardcode the background option and check for properties[p].unique
```

Comment:
Really depends on how you like your ternary conditionals indented, but point taken. I'll change it to a less....controversial indentation.